File: solutions/ominous-omino/java/approach2.java
import java.util.Scanner;


public class approach2
{
    static int N, R, C;

    public static void main(String ... orange) throws Exception
    {
        Scanner input = new Scanner(System.in);
        int t = input.nextInt();
        for (int n = 0; n < t; n++)
        {
            N = input.nextInt();
            R = input.nextInt();
            C = input.nextInt();
            boolean win=winner();
            if(win)
                System.out.printf("Case #%d: %s\n", n + 1,"GABRIEL");
            else
                System.out.printf("Case #%d: %s\n", n + 1,"RICHARD");
        }
    }

    static boolean winner()
    {
        if (R * C % N != 0)
            return false;

        if (N >= 7)
            return false;
        else if (N == 6)
        {
            if (Math.min(R, C) <= 3)
                return false;
        }
        else if (N == 5)
        {
            if (Math.min(R, C) <= 2 ||
                    Math.min(R, C) == 3 && Math.max(R, C) == 5)
                return false;
        }
        else if (N == 4)
        {
            if (Math.min(R, C) <= 2)
                return false;
        }
        else if (N == 3)
        {
            if (Math.min(R, C) <= 1)
                return false;
        }
        return true;
    }
}

File: solutions/ominous-omino/go/approach2.go
package main

import "fmt"

func winner(x, r, c int) bool {
	if r > c {
		r, c = c, r
	}
	if r * c % x != 0 {
		return true
	}
	if x <= 2 {
		return false
	}
	if x >= 7 {
		return true
	}
	if c < x {
		return true
	}
	if (x + 1) / 2 > r {
		return true
	}
	if x == 3 {
		return false
	}
	if x == 4 {
		return r == 2
	}
	if x == 5 {
		return r == 3 && c < 10
	}
	if x != 6 {  }
	return r == 3
}

func main() {
   var T int
   fmt.Scanln(&T)
   if T < 1 || T > 100 { panic(T) }
   for t := 1; t <= T; t++ {
		var X, R, C int
		fmt.Scanln(&X, &R, &C)
      fmt.Printf("Case #%d: ", t)
		if winner(X, R, C) {
			fmt.Println("RICHARD")
		} else {
			fmt.Println("GABRIEL")
		}
   }
}